File: backend/slotify/spring/src/main/java/com/slotify/backend/spring/service/dtos/ServiceSummary.java
package com.slotify.backend.spring.service.dtos;

import lombok.Builder;
import lombok.Getter;
import lombok.ToString;

import java.util.List;
import java.util.UUID;

@Builder
@Getter
@ToString
public class ServiceSummary {

    private UUID serviceId;
    private String serviceName;
    private Integer serviceMinutesDuration;
    private Integer servicePriceCent;
    private String s3ImageKey;
    private String description;

    private List<ScheduleSummary> schedules;
}

File: frontend/app/src/main/java/com/raj/slotify/dtos/service/ServiceSummary.kt
package com.raj.slotify.dtos.service

import com.google.gson.annotations.SerializedName
import java.util.UUID

data class ServiceSummary(
    @SerializedName("serviceId")
    val serviceId: UUID,

    @SerializedName("serviceName")
    val serviceName: String,

    @SerializedName("serviceMinutesDuration")
    val serviceMinutesDuration: Int,

    @SerializedName("servicePriceCent")
    val servicePriceCent: Int,

    @SerializedName("s3ImageUrl")
    val s3ImageUrl: String?,

    @SerializedName("description")
    val description: String?,

    @SerializedName("schedules")
    val schedules: List<ScheduleSummary>
)
